### ISSUE TITLE
Hadoop Hive support

### ISSUE THREAD

**Issue by ghost** [flyway/flyway#142]
_Original author: christia...@gmail.com (October 28, 2011 20:18:52)_

Database support for Hadoop Hive

_Original issue: http://code.google.com/p/flyway/issues/detail?id=172_

**Comment by TonyHolroyd** [comment]
+1

**Comment by varadharajan** [comment]
+1

**Comment by andr83** [comment]
+1

**Comment by eljefewatson** [comment]
I see this is a pretty old request, but I poked around in the source code for sqllite, and the effort to add support for another database doesn't look bad at all (I should probably poke at some of the more complicated databases to make sure).  I'll look into trying do a build and see if I can get it running from source and go from there.   We're using Hortonworks, but being Apache, hopefully will work for CDH, MapR, etc.

**Comment by mikaelv** [comment]
I am working on this, see my fork here: https://github.com/mikaelv/flyway
I have most of the unit tests working

**Comment by eljefewatson** [comment]
Think it's good enough to try yet?

**Comment by mikaelv** [comment]
Yes, feel free to try it. I had used a less mature implementation on a production schema with no particular issue. The flyway repair command requires some specific hive settings though (see hive-site.xml in my Pull Request), because it has to UPDATE the schema table.

**Comment by mambistats** [comment]
Do you think, you can make a binary release for the hive support? I see that your pull request is stilll pending. I am trying to setup the build, but there are so many dependencies on other DBs JDBC Jars ... 

May be it would be a good idea to add a script to the build, which loads the necessary JDBC Jars from the Internet. I understand, that due to the license issues it is impossible to add them to the GitHub Repo or the Maven Repos ... but I think one could provide a shell script or soemthing similiar, that downloads all dependencies which cannot be downloaded from Maven ...

**Comment by eljefewatson** [comment]
I never got my flyway build working in the 1st place, so sadly, the answer is no.  In the interim what I was really looking for was a way to build the Hive SQL scripts via tool rather than by hand.  Flyway is a nicety rather than a necessity.  I'll plug the tool DbSchema for it's ability to work with Hive and supposedly HBase using Phoenix because all they really need is a JDBC driver.  They can suck in schemas, tables, and column types based on the info provided by the JDBC driver.  They also can generate SQL between their project and the database or between projects to build both a database SQL baseline (when new developers need to build their database from scratch) and build update SQL scripts for test and production environments.  I can customize the SQL to match what Hive and Phoenix require.  And they fix bugs within a day.  ***end soapbox***

**Comment by mambistats** [comment]
Flyway has big architectural flaw in my mind ... they depend on the brand of the database. 

I prefer a tool which is happy when it can connect with JDBC and it should simply execute the SQL I write into my migration scripts.

And finally the tool should me allow to use a CSV File or table in a second JDBC connection (other DB) to store the Schema Version Info.

And suprise: you can do that with liquibase in Offline Database Mode using a so called unsupported Database, checking changes against Schema Version Info in an CSV file wrting the resulting SQL for the update to stdout and the new Schema Version info into the CSV file. This is perfect! Exactly what I need to migrate my Hive DBs. This is a smart architecture to easily support unknown Brand and SQL Modes.

Now I am back to liquibase.

**Comment by codingtony** [comment]
+1

**Comment by wilderrodrigues** [comment]
+1 I need this as well!

**Comment by CyborTronik** [comment]
+1

**Comment by gmcoringa** [comment]
@varuniteam my solution is not complete, and I think its very hard to add support to hive with the actual flyway design, because hive is not a relational database, as such do not support transactions, which is used in many parts by flyway.

**Comment by iglezouton** [comment]
I guess that using the HiveMetastore (Relation Database) directly is a better aproach... WDYT?
